genai_cookbook/nbs/2-fundamentals-unstructured-chain.md

```diff
@@ -17,6 +17,6 @@ Once the data has been processed by the data pipeline, it is suitable for use in
 
 5. **(Optional) Post-processing:** The LLM's response may be processed further to apply additional business logic, add citations, or otherwise refine the generated text based on predefined rules or constraints.
```

Comment:
Hm, we should probably also update the content here - stuff like "The prompt that will be sent to the LLM is formed by augmenting the user’s query with the retrieved context, in a template that instructs the model how to use each component, often with additional instructions to control the response format. The process of iterating on the right prompt template to use is referred to as prompt engineering." is not accurate. Let's see how much content we have to update/how much work that is..

Reply:
Like the actual steps in an agentic workflow are 1. the agent LLM tries to understand the user's query 2. the agent LLM determines how to respond (whether or not to call a tool) or simply generate a final response, 3. agent logic runs the tool and returns outputs back to the agent LLM, 4. the agent logic queries the agent LLM with the additional context of the tool output, and the agent LLM returns to step 2 etc